src/block_neural.rs

```diff
@@ -213,7 +213,7 @@ impl<L: OptimizerTrait + 'static> BlockNeuronLayer<L> {
                 alpha,                              //   alpha: f32,
                 self.weights.get_unchecked(0..),    //  a: &[f32],
                 self.num_inputs as i32,             //lda: i32,
-                &input_tape.get_unchecked(0..),     //   x: &[f32],
+                input_tape.get_unchecked(0..),      //   x: &[f32],
```

Comment:
Perhaps worth checking if comments still reflect the actual arg space (otherwise they could be misleading)

Reply:
They still do, this was actually &&[f32], which got directly dereferenced by the compiler to &[f32]